src/lib/handler/auth.api.ts

```diff
@@ -0,0 +1,26 @@
+import { User } from '@prisma/client';
+import { StatusCodes } from 'http-status-codes';
+import { NextResponse, NextRequest } from 'next/server';
+import { getToken } from 'next-auth/jwt';
+
+import * as userRepository from '@/repositories/user';
+
+export default async function appHandler(
+  request: NextRequest,
+  callback: (currentUser: User) => Promise<NextResponse>
+) {
+  try {
+    const token = await getToken({ req: request });
+    const currentUser = await userRepository.findById(token?.userId as number);
+
+    if (!currentUser)
+      return NextResponse.json({ message: 'Invalid token' }, { status: StatusCodes.UNAUTHORIZED });
```

Comment:
For multiline conditional, always use parentheses:

```suggestion 
     if (!currentUser)
         return NextResponse.json({ message: 'Invalid token' }, { status: StatusCodes.UNAUTHORIZED });
     }
```